### PR TITLE
Nerf "magical" battery storage options

### PR DESCRIPTION
**Pull request by Coolthulhu**
Makes tools with double battery mod not save any volume in storage. The double battery mod is an extra battery compartment added on top of the regular one, it shouldn't allow packing batteries denser than 100 charges per 1 volume under no circumstances.

Made tazer rechargeable and thus removed the ability to unload it.
Other options would be to make it volume 5 (too huge), limit it to 1 use when fully charged (big nerf) or lower its battery usage and maximum charge with it (could be considered too, now that they aren't instant stunlock against hulks).

Powered quarterstaff explicitly uses a tazer, so it gets the same changes as the tazer itself.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -2432,11 +2432,17 @@ int item::volume( bool integral ) const
...
-// tool mods also add volume
-    if (has_flag("ATOMIC_AMMO")) {
+    // Battery mods also add volume
+    if( has_flag("ATOMIC_AMMO") ) {
         ret += 1;
     }
 
+    if( has_flag("DOUBLE_AMMO") ) {
+        // Batteries have volume 1 per 100 charges
+        // TODO: De-hardcode this
```
Comment (mugling) [inline src/item.cpp]: Set `stack_size 100` in `ammo.json` and then derive it as `ammo_data()->ammo->stack_size`.

Comment (Coolthulhu) [inline src/item.cpp]:
```
/** Specific ammo data, returns nullptr if item is neither ammo nor loaded with any */
const itype * ammo_data() const;
```

`nullptr` for unloaded items.
I'd need to add a special case just for unloaded items and that could result in discrepancies where a loaded item could lose volume.

Comment (mugling) [inline src/item.cpp]: `ammo_data() ? ammo_data()->id : default_ammo( ammo_type() )` or something to that effect

Comment (Coolthulhu) [inline src/item.cpp]: > I'd need to add a special case just for unloaded items and that could result in discrepancies where a loaded item could lose volume.

Comment (mugling) [inline src/item.cpp]: or even `default_ammo( ammo_type() )` which avoids the special case and allows `DOUBLE_AMMO` to be applied to items that take ammo other than batteries?

**Comment by Rivet-the-Zombie** [comment]
>or lower its battery usage and maximum charge with it

I like this idea. The stun effect is weak enough that it hardly seems like it's worth 100 batteries.

**Pull request merged**